chia/server/start_introducer.py

```diff
@@ -14,37 +16,54 @@
 SERVICE_NAME = "introducer"
 
 
-def service_kwargs_for_introducer(
+def create_introducer_service(
     root_path: pathlib.Path,
     config: Dict,
-) -> Dict:
+    advertised_port: Optional[int] = None,
+    connect_to_daemon: bool = True,
+) -> Service:
     service_config = config[SERVICE_NAME]
 
+    if advertised_port is None:
+        advertised_port = service_config["port"]
+
     introducer = Introducer(service_config["max_peers_to_send"], service_config["recent_peer_threshold"])
     node__api = IntroducerAPI(introducer)
     network_id = service_config["selected_network"]
-    kwargs = dict(
+    return Service(
         root_path=root_path,
         config=config,
         node=introducer,
         peer_api=node__api,
         node_type=NodeType.INTRODUCER,
-        advertised_port=service_config["port"],
         service_name=SERVICE_NAME,
         server_listen_ports=[service_config["port"]],
         network_id=network_id,
+        advertised_port=advertised_port,
+        connect_to_daemon=connect_to_daemon,
     )
-    return kwargs
 
 
-def main() -> None:
+async def async_main() -> int:
     # TODO: refactor to avoid the double load
     config = load_config(DEFAULT_ROOT_PATH, "config.yaml")
     service_config = load_config_cli(DEFAULT_ROOT_PATH, "config.yaml", SERVICE_NAME)
     config[SERVICE_NAME] = service_config
-    kwargs = service_kwargs_for_introducer(DEFAULT_ROOT_PATH, config)
-    return run_service(**kwargs)
+    service = create_introducer_service(DEFAULT_ROOT_PATH, config)
+    initialize_logging(
+        service_name=SERVICE_NAME,
+        logging_config=service_config["logging"],
+        root_path=DEFAULT_ROOT_PATH,
+    )
+    await service.setup_process_global_state()
+    await service.run()
+
+    return 0
+
+
+def main() -> int:
+    return async_run(async_main())
 
 
 if __name__ == "__main__":
-    main()
+    sys.exit(main())
```

Comment:
```suggestion
    sys.exit(async_run(main()))
```

Why not drop the `def main() -> int` above, rename `async_main` to `main` and do the above in all files?

Reply:
>  then the entry points reminded me that we need a sync callable 

Not sure what you mean here, don't we have our sync callable with `async_run` which gets used in the entry point and all the other fun we want to do can just go into the async `main` then or why do we need that one to be sync too?  

```python
async def main() -> int:
    return 0


if __name__ == "__main__":
    sys.exit(async_run(main()))
```